### PR TITLE
add feat pagiantion,filter by status, specify date, order by date

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
[- ticket redmine](https://edu-redmine.sun-asterisk.vn/issues/74343)

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
image

image

### CODE REVIEW

**Comment by ghost** [comment]
ready

File: src/controller/admin/admin.controller.ts [hunk 1]
```diff
@@ -1,32 +1,91 @@
...
 
+interface MyQueryParams {
+  sortASC?: string;
+  status?: string;
+  date?: string;
+  minRange?: string;
+  maxRange?: string;
+  page?: number;
+  limit?: number;
+}
```
Comment (huongnt-2545) [inline src/controller/admin/admin.controller.ts:23]: cái này e thay thế dùng DTO đc nè

File: src/service/invoice.service.ts [hunk 2]
```diff
@@ -146,40 +145,94 @@ export class InvoiceService {
...
+
+  public async getOrders(options: {
+    status?: STATUS_ORDER[] | undefined;
+    startDate?: Date | undefined;
+    endDate?: Date | undefined;
+    sortASC?: boolean | undefined;
+    minRange?: number | undefined;
+    maxRange?: number | undefined;
+    page?: number;
+    limit?: number;
```
Comment (huongnt-2545) [inline src/service/invoice.service.ts]: sau khi thay thế interface bằng DTO thì chõ này em define options có type là class dto nha

**Comment by ghost** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Comment by huongnt-2545** [comment]
merged

**Pull request merged**